### ISSUE TITLE
Sidekiq entreprise enabling/disabling cron in UI

### ISSUE THREAD

**Issue by mathieusanchez** [mperham/sidekiq#5000]
Hey,

We are using Sidekiq enterprise and we just upgraded from sidekiq-ent 1 to 2.2.3.
Before, we were using the [sidekiq-cron](https://github.com/ondrejbartas/sidekiq-cron) to be able to add crons on our platform.
But now we want to use your Periodic Jobs but we are seeing that we are getting regression on one functionality.

With the external gem we were to enable or disable any cron from the sidekiq web interface, but with your Peridic Jobs it isn't possible.

Are you planning to add this functionality?
If yes do you have an approximate date?

**Comment by mperham** [comment]
You want to "pause" a given cron job? Seems reasonable.

The problem is that there is no unique identifier for a given cron. I'm not sure how to work around this.

**Comment by mathieusanchez** [comment]
Yes, what we would like is to be able to pause a cron from the UI.

Do you think you can add it ?

**Comment by rainerborene** [comment]
It would be possible to implement a `stoppable` abstraction that could be reused on `batches`, `cron_jobs` and `queues`?

**Comment by mperham** [comment]
I can't implement anything without a unique name for the thing. Jobs have JID. Batches have BID. Cron jobs have ???

**Comment by rainerborene** [comment]
@mperham  Cron jobs have CID :rofl: 

But seriously, it would be possible to implement stoppable cron jobs using the dynamic jobs approach which is explained on the [official documentation](https://github.com/mperham/sidekiq/wiki/Ent-Periodic-Jobs#dynamic-jobs). So, there is no need to implement this IMO. I think you should consider implementing it for `batches` though.

**Comment by mperham** [comment]
It is implemented for batches, as invalidation.  You can’t pause a batch but you can invalidate it.

**Comment by mathieusanchez** [comment]
@mperham so you cannot implement a pause option on cron ?

**Comment by mperham** [comment]
I've already stated the problem.

> The problem is that there is no unique identifier for a given cron. I'm not sure how to work around this.

**Comment by mperham** [comment]
Do you see how each cron in sidekiq-cron has a name, which must be unique? Enterprise has no such thing.

**Comment by mperham** [comment]
Sorry, I closed the issue too early. Yes, I like the idea and I can implement it but it's not easy; I will need to think about how best to do it without a unique name.

**Comment by adamniedzielski** [comment]
@mperham `#{cron_expression}-#{worker_class}` is going to be a unique identifier for 99.9% of cases. Probably there are some valid cases to run the same job at exactly the same schedule, but with different options. That said, if you're comfortable with not being 100% correct, this could be solved with a comment in the Wiki - https://github.com/mperham/sidekiq/wiki/Ent-Periodic-Jobs

**Comment by hbontempo-br** [comment]
Hi! Would love to have this feature, specially if exposed through Sidekiq's API as well.

If a unique identifier is required why not ask for the the for it explicitly? Like a "cron identifier".
For backward compatibility something like @adamniedzielski could work (maybe something more strict like a MD5 of the `args` and `kargs` of the schedule registration).

**Comment by lusinh** [comment]
Why `https://github.com/sidekiq-scheduler/sidekiq-scheduler` can do a thing such as enable/disable, run now scheduler like cron job but enterprise version cannot?

**Comment by danilo-silva-cw** [comment]
Hi guys, I created this gem that adds pause and unpause functionality to sidekiq-enterprise. 

https://github.com/dannnylo/sidekiq-belt/tree/main?tab=readme-ov-file#pause-periodic-jobs-sidekiq-enterprise
https://rubygems.org/gems/sidekiq-belt

**Comment by sobrinho** [comment]
@mperham those features are the only reason we still use sidekiq-cron instead of periodic jobs: pause and manual enqueue.

What I can think of is:

1. To "pause" a job, we let the job run — we don't mess with the scheduler. Before running the job class, we check a Redis bit/key/hset (or similar) and skip the execution, logging that it's paused.
2. To "trigger" a job, we simply enqueue it with the configuration to execute right now, as we would with `MyJob.perform_async(...)`.

I'm happy to write up a PR or send it to you privately given the closed source.

@danilo-silva-cw's approach is great, but I still believe pause/trigger should be native and not a 3rd party gem.

**Comment by mperham** [comment]
Cron pause/enqueue functionality via Web UI has been available since 8.0.1.

https://github.com/sidekiq/sidekiq/wiki/Ent-Periodic-Jobs#web-ui

**Comment by sobrinho** [comment]
Glad to hear. We are yet stuck on 7.2.4, I will take a try once it's upgraded! 🥳
